### PR TITLE
MODKBEKBJ-429-Support POST /eholdings/loading/kb-credentials/{credentialsId} to load holdings for certain KB Credentials	

### PR DESCRIPTION
**Pull request by natali-zaitseva**
## Purpose
Support functionality of load holdings by credentials id described in https://issues.folio.org/browse/MODKBEKBJ-429, which is a part of 
[Single tenant spike](https://wiki.folio.org/pages/viewpage.action?pageId=36572764)

## Approach
* modified ModuleDescriptor file by adding a new endpoint
* updated raml file with description for new endpoint POST /eholdings/loading/kb-credentials/{credentialsId}
* updated existing implementation by adding credentials id as a parametr
* partially included implementation for stories 
 ** https://issues.folio.org/browse/MODKBEKBJ-427 - Update GET /loadHoldings/status to support getting status for certain credentials
 ** https://issues.folio.org/browse/MODKBEKBJ-428 - Set loading status "Not Started" on KB Credentials creation

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/integrationsuite/EholdingsTitlesTest.java [hunk 1]
```diff
@@ -115,7 +119,7 @@
   @Override @Before
   public void setUp() throws Exception {
     super.setUp();
-    setupDefaultKBConfiguration(getWiremockUrl(), vertx);
+    insertKbCredentials(STUB_CREDENTILS_ID, getWiremockUrl(), STUB_CREDENTIALS_NAME, STUB_API_KEY, STUB_CUSTOMER_ID, vertx);
```
[the remark this answers is not in the document]
Comment (natali-zaitseva) [reply src/test/java/org/folio/rest/impl/integrationsuite/EholdingsTitlesTest.java]: fixed

File: src/main/java/org/folio/rest/impl/LoadHoldingsImpl.java [hunk 2]
```diff
@@ -65,8 +67,30 @@ public void getLoadHoldingsStatus(String contentType, Map<String, String> okapiH
...
+
+  @Override
+  @Validate
+  @HandleValidationErrors
+  public void postEholdingsLoadingKbCredentialsById(String id, String contentType, Map<String, String> okapiHeaders,
+                                                    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    logger.info("Start loading of holdings for credentials: " + id);
+    RMAPITemplate template = templateFactory.createTemplate(okapiHeaders, asyncResultHandler);
+    template.requestAction(context ->
+      credentialsService.findById(id, okapiHeaders)
```
[the remark this answers is not in the document]
Comment (natali-zaitseva) [reply src/main/java/org/folio/rest/impl/LoadHoldingsImpl.java]: fixed, thanks

File: src/main/java/org/folio/rest/impl/LoadHoldingsImpl.java [hunk 3]
```diff
@@ -65,8 +67,30 @@ public void getLoadHoldingsStatus(String contentType, Map<String, String> okapiH
     String tenantId = TenantTool.tenantId(okapiHeaders);
     RMAPITemplate template = templateFactory.createTemplate(okapiHeaders, asyncResultHandler);
     template
-      .requestAction(context -> holdingsStatusRepository.get(tenantId))
+      .requestAction(context ->
+        credentialsService.findByUser(okapiHeaders)
+          .thenCompose(kbCredentials -> holdingsStatusRepository.findByCredentialsId(kbCredentials.getId(), tenantId))
```
[the remark this answers is not in the document]
Comment (natali-zaitseva) [reply src/main/java/org/folio/rest/impl/LoadHoldingsImpl.java]: fixed

File: src/main/resources/templates/db_scripts/create-holdings-status-audit-table.sql [hunk 4]
```diff
@@ -1,4 +1,11 @@
...
-ADD COLUMN IF NOT EXISTS operation VARCHAR(20) NOT NULL,
-ADD COLUMN IF NOT EXISTS updated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP;
+  DROP COLUMN IF EXISTS id,
+  ADD COLUMN IF NOT EXISTS credentials_id UUID NOT NULL,
+  ADD COLUMN IF NOT EXISTS operation VARCHAR(20) NOT NULL,
+  ADD COLUMN IF NOT EXISTS updated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP;
+
+ALTER TABLE holdings_status_audit
+  DROP CONSTRAINT IF EXISTS fk_holdings_status_audit_kb_credentials,
+  ADD CONSTRAINT fk_holdings_status_audit_kb_credentials FOREIGN KEY (credentials_id) REFERENCES kb_credentials (id);
```
Comment (dmtkachenko) [inline src/main/resources/templates/db_scripts/create-holdings-status-audit-table.sql:9]:
```suggestion
  ADD CONSTRAINT fk_holdings_status_audit_kb_credentials FOREIGN KEY (credentials_id) REFERENCES kb_credentials (id) ON DELETE CASCADE;
```

Comment (dmtkachenko) [reply src/main/resources/templates/db_scripts/create-holdings-status-audit-table.sql:9]: .. and for the rest of holdings loading tables

Comment (natali-zaitseva) [reply src/main/resources/templates/db_scripts/create-holdings-status-audit-table.sql]: fixed, thanks

**Pull request merged**